tests/kafkatest/tests/core/upgrade_test.py

```diff
@@ -62,6 +62,7 @@ def perform_upgrade(self, from_kafka_version, to_message_format_version=None):
             self.kafka.start_node(node)
 
     @cluster(num_nodes=6)
+    @parametrize(from_kafka_version=str(LATEST_0_10_1), to_message_format_version=None, compression_types=["lz4"])
```

Comment:
Maybe we should have a line for the old consumer as well.

Regarding your question about the cluster id, the following line needs to be conditional on `from_kafka_version <= LATEST_0_10_0`:

```
assert self.zk.query("/cluster/id") is None
```

The code is just checking that the cluster id is generated after an upgrade from 0.10.0.x or lower to 0.10.1x or higher.